### PR TITLE
FOGL-647 configuration manager with storage engine

### PR DESCRIPTION
**Pull request by ashish-ScaleDB**
- [x] **configuration manager**

    - [x]  /foglamp/categories
    - [x]  /foglamp/category/{category_name}
    - [x]  /foglamp/category/{category_name}/{config_item}
    - [x]  /foglamp/asset/{category_name}/{config_item}/value

Note: 

1) configuration manager writer - **ConfigurationManager(object)**
2) API tests:

```
 collected 8 items 

tests/core/api/test_configuration.py ......s.

============ 7 passed, 1 skipped in 0.19 seconds ======
````

3) Unit tests: 

```
collected 25 items 

tests/test_configuration_manager.py ...s.............X.xxX...

=========== 20 passed, 1 skipped, 2 xfailed, 2 xpassed in 1.12 seconds ========
```
`Added and skipped  1 test about FOGL-690`

~4) **Configuration reference removal for all services - PENDING**~

### CODE REVIEW

File: src/python/foglamp/configuration_manager.py [hunk 1]
```diff
@@ -147,22 +154,32 @@ def _run_callbacks(category_name):
...
+    # configuration.ts FROM configuration WHERE configuration.key = :key_1
+    # payload = PayloadBuilder().SELECT(("value"))\
+    #     .WHERE(["key", "=", category_name]).payload()
+    # _storage = connect.get_storage()
+    # results = _storage.query_tbl_with_payload('configuration', payload)
+    # for row in results['rows']:
+    #     return row['value']
     async with aiopg.sa.create_engine(_connection_string) as engine:
         async with engine.acquire() as conn:
             async for row in conn.execute(
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/configuration_manager.py]: Kindly advise!

Comment (MarkRiddoch) [reply src/python/foglamp/configuration_manager.py]: The code in the comment looks right to be, I would added a return item to limit what is returned, but this is not a requirement. E.g. 
{
	"where" : {
			"column" : "key",
 			"condition" : "=",
			"value" : "SCHEDULER"
		},
	"return" : [ "value" ]
}

Where the key is SCHEUDLER

**Comment by ashwinscale** [comment]
I see issues on startup. Please let me know how to successfully run it.
(us1704) ususer@us1704:~/fogl518/FogLAMP/src/python$ python3 -m foglamp.core
DEV 0.0.0.0 33585
DEV 0.0.0.0 33585
DEV 0.0.0.0 33585
Storage Service:  <class 'foglamp.storage.storage.Storage'>
Error: 'NoneType' object is not subscriptable

Nov  6 11:09:55 us1704 [FOGLAMP] 11-06-2017 11:09:55 - INFO :: server: foglamp.core.server: start core
Nov  6 11:09:55 us1704 [FOGLAMP] 11-06-2017 11:09:55 - INFO :: server: foglamp.core.server: Management API started on http://0.0.0.0:33585
Nov  6 11:09:55 us1704 [FOGLAMP] 11-06-2017 11:09:55 - INFO :: server: foglamp.core.server: start storage, from directory /home/ususer//services/storage
Nov  6 11:09:55 us1704 [FOGLAMP] 11-06-2017 11:09:55 - INFO :: server: foglamp.core.server: start scheduler
Nov  6 11:09:55 us1704 [FOGLAMP] 11-06-2017 11:09:55 - INFO :: scheduler: foglamp.core.scheduler.scheduler: Starting
Nov  6 11:09:55 us1704 port[1818]: Load storage plugin postgres.
Nov  6 11:09:55 us1704 port[1818]: Loaded storage plugin postgres.
Nov  6 11:09:55 us1704 port[1818]: Starting management api on port 0.
Nov  6 11:09:55 us1704 port[1818]: Starting service...
Nov  6 11:10:05 us1704 [FOGLAMP] 11-06-2017 11:10:05 - INFO :: instance: foglamp.microservice_management.service_registry.instance: Registered service instance id=3b871f97-edbd-4e9d-814c-d7bdfd9150cc: <FogLAMP Storage, type=Storage, protocol=http, address=localhost, service port=42893, management port=37051, status=0>
Nov  6 11:10:05 us1704 t[1818]: Registered service 3b871f97-edbd-4e9d-814c-d7bdfd9150cc.
Nov  6 11:10:10 us1704 [FOGLAMP] 11-06-2017 11:10:10 - INFO :: scheduler: foglamp.core.scheduler.scheduler: Starting Scheduler: Management port received is 33585
Nov  6 11:10:10 us1704 [FOGLAMP] 11-06-2017 11:10:10 - ERROR :: storage: foglamp.storage.storage: Client error code: 400

**Comment by praveen-garg** [comment]
@ashish-ScaleDB Please re-test FOGL-690 and enable the unit test; may be bring back scheduler.py `running` > `'running'` also to ensure without tests; if still does not work lt test fail but allow scheduler to run; as its just text info!

File: src/python/foglamp/core/scheduler/scheduler.py [hunk 2]
```diff
@@ -662,17 +662,17 @@ def _schedule_first_task(self, schedule, current_time):
             },
             "max_completed_task_age_days": {
                 "description": "The maximum age, in days (based on the start time), for a rows "
-                               "in the tasks table that do not have a status of 'running'",
+                               "in the tasks table that do not have a status of running",
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/core/scheduler/scheduler.py]: @MarkRiddoch As @ashwinscale has pointed out notice the word 'running' in single quotes that will produce an error if it goes through storageclient-> storage service
```
{'message': 'ERROR:  syntax error at or near "running"\nLINE 1: ... in the tasks table that do not have a status of \'running\'",...\n                                                             ^\n', 'entryPoint': 'insert', 'retryable': False}
```

So I removed these quotes and now it is working fine. Also I tried with `\'running\'` and yes it also works with latest storage build (including FOGL-690 changes)

I have added one unit test regarding it and skipped with `FOGL-690` (tests.test_configuration_manager.TestConfigurationManager#test_create_category_with_quoted_json_data). So I thought that fix would be in FOGL-690 itself. 

Kindly advise!

**Pull request merged**